### PR TITLE
feat(iOS): replace fbsimctl with AppleSimUtils for `setLocation`.

### PR DESCRIPTION
**Pull request by asafkorem**
Replace deprecated fbsimctl with AppleSimulatorUtils for `setLocation` API.

AppleSimUtils `setLocation` is now working as expected thanks to a workaround introduced in [version 0.9.6](https://github.com/wix/AppleSimulatorUtils/releases/tag/0.9.6) and another fix from [version 0.9.7](https://github.com/wix/AppleSimulatorUtils/releases/tag/0.9.7).

`setLocation` API (of ASU) requires from the Simulator app to run on non-headless mode (the Simulator app must be opened).

### CODE REVIEW

**Comment by asafkorem** [comment]
I think the only thing needed is to execute `xcode-select -s /Applications/Xcode.app` on our agents.
Otherwise we're getting the `Unable to find application named 'Simulator'` error, which prevents us from setting the location during tests (the Simulator app must be open while changing its location).

File: detox/index.d.ts [hunk 1]
```diff
@@ -600,13 +600,9 @@ declare global {
...
-             *
-             * <p/>On iOS `setLocation` is dependent on [fbsimctl](https://github.com/facebook/idb/tree/4b7929480c3c0f158f33f78a5b802c1d0e7030d2/fbsimctl)
-             * which [is now deprecated](https://github.com/wix/Detox/issues/1371).
-             * If `fbsimctl` is not installed, the command will fail, asking for it to be installed.
-             *
-             * <p/>On Android `setLocation` will work with both Android Emulator (bundled with Android development tools) and Genymotion.
-             * The correct permissions must be set in your app manifest.
+             * - On iOS, `setLocation` is dependent on AppleSimulatorUtils, that has `--setLocation` support.
+             * - On Android, `setLocation` will work with both Android Emulator (bundled with Android development tools) and Genymotion.
+             *   The correct permissions must be set in your app manifest.
```
Comment (d4vidi) [inline detox/index.d.ts:605]: 🚀

File: detox/src/devices/common/drivers/ios/tools/AppleSimUtils.js [hunk 2]
```diff
@@ -267,16 +267,30 @@ class AppleSimUtils {
...
+      await this._openSimulatorApp(udid);
+
+      await this._execAppleSimUtils({ args: `--byId ${udid} --setLocation "[${lat}, ${lon}]"` });
+    } catch (e) {
+      const stderr = e && e.stderr || '';
+
+      if (stderr.match(/Unknown command line option.*--setLocation/)) {
+        throw new DetoxRuntimeError({
+          message: `Failed to set the location (${lat}, ${lon}) on the device ${udid}.`
+          + '\nYour current "applesimutils" version needs to be upgraded to 0.9.3 or higher.',
```
Comment (d4vidi) [inline detox/src/devices/common/drivers/ios/tools/AppleSimUtils.js:281]: "needs to be upgraded to the latest version." 🤔

Comment (d4vidi) [inline detox/src/devices/common/drivers/ios/tools/AppleSimUtils.js:281]:
```suggestion
          + '\nYour current "applesimutils" version needs to be upgraded to the latest version',
```

File: detox/test/e2e/16.location.test.js [hunk 3]
```diff
@@ -1,45 +1,61 @@
...
-//TODO: Ignoring the test in CI until fbsimctl supports Xcode 9
-async function isFbsimctlInstalled() {
-  try {
-    await exec(`which fbsimctl`);
-    return true;
-  } catch (e) {
-    console.log(`setLocation only works through fbsimctl currently`);
-    return false;
-  }
-}
```
Comment (d4vidi) [inline detox/test/e2e/16.location.test.js:12]: YES!

File: detox/test/e2e/16.location.test.js [hunk 4]
```diff
@@ -1,45 +1,61 @@
...
+  });
 
-    await expect(element(by.text(`Latitude: ${lat}`))).toBeVisible();
-    await expect(element(by.text(`Longitude: ${lon}`))).toBeVisible();
+  it('should allow to get and set location', async () => {
+    await expectLocationToBeAvailable();
   });
 });
+
+// Helpers
```
Comment (d4vidi) [inline detox/test/e2e/16.location.test.js:43]: The typical pattern is to use a _driver_. Example: https://github.com/wix/Detox/blob/master/detox/test/e2e/drivers/actions-driver.js

Comment (asafkorem) [reply detox/test/e2e/16.location.test.js:43]: done :+1:

**Comment by asafkorem** [comment]
Still need to figure out how we can overcome the fact that we can't open the Simulator app on CI @d4vidi

**Comment by asafkorem** [comment]
Update: fb/idb's (formerly fbsimctl) set-location API is not working on headless (without open GUI) Simulator mode as well.

**Comment by jpbalo** [comment]
@asafkorem can we get this released?

**Comment by asafkorem** [comment]
Not stale, I think we should get back to this PR soon 🙂

**Comment by Alaa-Ben** [comment]
Hi !

Not sure how helpful there is, but here's how I'm getting the location to work on IOS (headless mode)
Just dropping this here in case of !
If it doesn't help here, it might help people who are waiting for this PR i guess :D 

Capture d’écran 2022-10-04 à 08 48 42

**Comment by asafkorem** [comment]
> Hi !
> 
> Not sure how helpful there is, but here's how I'm getting the location to work on IOS (headless mode) Just dropping this here in case of ! If it doesn't help here, it might help people who are waiting for this PR i guess :D
> 
> Capture d’écran 2022-10-04 à 08 48 42

Thank @Alaa-Ben! Sounds like a great contribution to Detox, if you'd like to take that to a PR.

**Comment by Alaa-Ben** [comment]
Sure ! It uses directly **idb** which is very reliable as it's used at facebook.
I'll try to get started on this next week 😄

**Pull request closed without merging**